### ISSUE TITLE
branche 0.17，after InnerProduct layers,BatchNorm got some error message,need help

### ISSUE THREAD

**Issue by 714142828** [NVIDIA/caffe#592]
I0125 15:23:57.180774 19274 layer_factory.hpp:172] Creating layer 'loss1/fc' of type 'InnerProduct'
I0125 15:23:57.180778 19274 layer_factory.hpp:184] Layer's types are Ftype:FLOAT Btype:FLOAT Fmath:FLOAT Bmath:FLOAT
I0125 15:23:57.180795 19274 net.cpp:205] Created Layer loss1/fc (106)
I0125 15:23:57.180801 19274 net.cpp:582] loss1/fc <- loss1/conv/bn/sc
I0125 15:23:57.180810 19274 net.cpp:552] loss1/fc -> loss1/fc
I0125 15:23:57.329124 19274 net.cpp:270] Setting up loss1/fc
I0125 15:23:57.329149 19274 net.cpp:277] TEST Top shape for layer 106 'loss1/fc' 1 1024 (1024)
I0125 15:23:57.329174 19274 net.cpp:138] Setting types for Layer loss1/fc/bn
I0125 15:23:57.329181 19274 layer_factory.hpp:172] Creating layer 'loss1/fc/bn' of type 'BatchNorm'
I0125 15:23:57.329188 19274 layer_factory.hpp:184] Layer's types are Ftype:FLOAT Btype:FLOAT Fmath:FLOAT Bmath:FLOAT
I0125 15:23:57.329221 19274 net.cpp:205] Created Layer loss1/fc/bn (107)
I0125 15:23:57.329226 19274 net.cpp:208] Test Layer for songbo loss1/fc/bn (107)
I0125 15:23:57.329236 19274 net.cpp:582] loss1/fc/bn <- loss1/fc
I0125 15:23:57.329257 19274 net.cpp:552] loss1/fc/bn -> loss1/fc/bn
F0125 15:23:57.329299 19274 blob.hpp:251] Check failed: axis_index < num_axes() (2 vs. 2) axis 2 out of range for 2-D Blob with shape 1 1024 (1024)
*** Check failure stack trace: ***
    @     0x7f2a502710cd  google::LogMessage::Fail()
    @     0x7f2a50272f33  google::LogMessage::SendToLog()
    @     0x7f2a50270c28  google::LogMessage::Flush()
    @     0x7f2a50273999  google::LogMessageFatal::~LogMessageFatal()
    @     0x55e1c33e3243  caffe::Blob::CanonicalAxisIndex()
    @     0x55e1c33e2fba  caffe::Blob::shape()
    @     0x7f2a5186b51e  caffe::BatchNormLayer<>::LayerSetUp()
    @     0x7f2a51a54794  caffe::CuDNNBatchNormLayer<>::LayerSetUp()
    @     0x7f2a516977c6  caffe::LayerBase::SetUp()
    @     0x7f2a5168af46  caffe::Net::Init()
    @     0x7f2a5168928b  caffe::Net::Net()
    @     0x55e1c33db167  test()
    @     0x55e1c33df829  main
    @     0x7f2a4e98cbf7  __libc_start_main
    @     0x55e1c33d829a  _start

**Comment by drnikolaev** [comment]
@714142828 looks like your model is not well formed. Could you attach complete log?

**Comment by 714142828** [comment]
> @714142828 looks like your model is not well formed. Could you attach complete log?

after I modified  src/caffe/layers/batch_norm_layer.cpp  , and remake caffe,my mode can work well ,thnaks 
  | 115 | int H = 1;
  | 116 | int W = 1;
  | 117 | if(bottom[0]->num_axes()>2){
  | 118 | H = bottom[0]->shape(2);
  | 119 | W = bottom[0]->shape(3);
  | 120 | }
